File: Q1_taskqueue.py
# Take input from the user
task_queue = input("Enter the task queue as a list: ").split()
current_index = int(input("Enter the current index: "))
inactive_members = input("Enter the inactive members as a list: ").split()

# Remove inactive members from the task queue
active_members = task_queue[:current_index+1]
for member in inactive_members:
    while member in active_members:
        active_members.remove(member)
active_members_set = set(active_members)
filtered_queue = [member for member in task_queue if member in active_members_set]

# Find the new index of the current member
current_member = filtered_queue[current_index % len(filtered_queue)]
new_current_index = filtered_queue.index(current_member)

# Output the results
print("Filtered queue after removing the inactive members are",filtered_queue)
print("New current index is",new_current_index)

File: Q1_taskqueue.java
import java.util.*;

public class Q1_taskqueue {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        // User inputs
        System.out.print("Enter the task queue as a list: ");
        List<String> taskQueue = Arrays.asList(scanner.nextLine().split(" "));
        System.out.print("Enter the current index: ");
        int currentIndex = Integer.parseInt(scanner.nextLine());
        System.out.print("Enter the active members as a list: ");
        List<String> activeMembers = Arrays.asList(scanner.nextLine().split(" "));

        // Filter task queue by active members
        Set<String> activeMembersSet = new HashSet<>(activeMembers);
        List<String> filteredQueue = new ArrayList<>();
        for (String member : taskQueue) {
            if (activeMembersSet.contains(member)) {
                filteredQueue.add(member);
            }
        }

        // Find the new index of the current member
        String currentMember = filteredQueue.get(currentIndex % filteredQueue.size());
        int newCurrentIndex = filteredQueue.indexOf(currentMember);

        // Output
        System.out.println("Filtered queue after filtering by active members are: " + filteredQueue);
        System.out.println("New current index is: " + newCurrentIndex);
    }
}

//Description
//We take task queue and current index and active members as an input as mentioned in the question
//
//Then A HashSet is created from the active members list to allow for fast membership testing. The program then iterates through 
//the task queue, and adds any member to the filtered queue if it is in the active members set

